### PR TITLE
docs: added warning about docker-sync usage

### PR DESCRIPTION
**Pull request by vitek-rostislav**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Developers should be aware of the possible problems when using docker-sync
|New feature| No 
|[BC breaks](https://github.com/shopsys/shopsys/blob/master/docs/contributing/backward-compatibility-promise.md)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| No

### CODE REVIEW

File: docs/installation/installation-guide.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 This document will provide you with information about 2 ways of developing and runninng the Shopsys Framework project and the services that it depends on.  
 The first option [using docker](#installation-using-docker) is **highly recommended** since it is the easiest and fastest way to start Shopsys Framework.
-In the case the operation system does not support docker, we prepared also second section with document about project [installation without docker](#installation-without-docker), however this way is slower and harder to configure and maintain because of different operation systems and their versions.
+In the case the operation system does not support docker, or you are not able to use Docker for some reason, we prepared also second section with document about project [installation without docker](#installation-without-docker), however this way is slower and harder to configure and maintain because of different operation systems and their versions.
```
Comment (boris-brtan) [inline docs/installation/installation-guide.md]: this seems to be a bit strange. please try to describe the real problem why would the developer want to not use it.

File: docs/installation/installation-using-docker-macos.md [hunk 2]
```diff
@@ -7,7 +7,9 @@ If you want to contribute to the framework itself,
 you need to install the whole [shopsys/shopsys](https://github.com/shopsys/shopsys) monorepo.
 Take a look at the article about [Monorepo](../introduction/monorepo.md) for more information.
 
-This solution uses [*docker-sync*](http://docker-sync.io/) (for fast two-way synchronization of the application files between the host machine and Docker volume).
+This solution uses [*docker-sync*](http://docker-sync.io/) (for relatively fast two-way synchronization of the application files between the host machine and Docker volume).
+
+**Warning: Docker-sync might be a burden for intensive project development, especially when you switch between branches or even between projects often. In such a case, you should consider using [native installation](./native-installation.md).**
```
Comment (boris-brtan) [inline docs/installation/installation-using-docker-macos.md]: please be more descriptive about the huge amount of files in shared volumes of virtualized docker. if the big project would have just small portion of files that are shared, intensive switching between branches should not be such problem

Comment (vitek-rostislav) [reply docs/installation/installation-using-docker-macos.md]: :thinking: but how can a huge project have shared just a small portion of files?

> huge amount of files in shared volumes of virtualized docker.

I would say that this state is by default after installing project-base, am I right?

File: docs/installation/installation-guide.md (hunk 1 above)
Comment (vitek-rostislav) [reply docs/installation/installation-guide.md]: I tried to rewrite it, is it better now?

File: docs/installation/installation-using-docker-macos.md (hunk 2 above)
Comment (vitek-rostislav) [reply docs/installation/installation-using-docker-macos.md]: However, I have tried to add the information there, is it better now?

**Pull request merged**